components/supervisor/pkg/supervisor/services.go

```diff
@@ -692,6 +704,38 @@ func (is *InfoService) WorkspaceInfo(context.Context, *api.WorkspaceInfoRequest)
 	return resp, nil
 }
 
+// getWorkspaceClass of current workspace
+func (is *InfoService) getWorkspaceClass(ctx context.Context) *api.WorkspaceInfoResponse_WorkspaceClass {
+	if is.gitpodService == nil {
+		gitpodService, err := createServerService(is.cfg, is.tokenService, []string{"function:getSupportedWorkspaceClasses"})
```

Comment:
we should not do it, service should be provided from supervisor.go if it is missing then no workspace class

there should be only one place where we create server service

Reply:
If we use the same server service, `git` token provider will not work because `function:getSupportedWorkspaceClasses` scope may not exists if server rollback.

Reply:
Let's provide info via env var like here [[1](https://github.com/gitpod-io/gitpod/blob/d3147bcb557cb7db1a3815f11239624f58335cc5/components/server/src/workspace/workspace-starter.ts#L1339-L1342)], i.e. pass `GITPOD_WORKSPACE_CLASS`? and then you will have it via `config.WorkspaceClass` if env var is missing we just return nil and client should respect it

No need for additional remote call or caching.